src/map/utils/battleutils.cpp

```diff
@@ -3089,6 +3096,14 @@ namespace battleutils
             * (100 + PAttacker->getMod(Mod::SKILLCHAINBONUS)) / 100
             * (100 + PAttacker->getMod(Mod::SKILLCHAINDMG)) / 100);
 
+        auto PChar = dynamic_cast<CCharEntity *>(PAttacker);
+        if (PChar && PChar->StatusEffectContainer->HasStatusEffect(EFFECT_INNIN))
+        {
+            auto angle = PDefender->loc.p.rotation - getangle(PDefender->loc.p, PChar->loc.p);
+            // assuming default tolerance of 42 from lua_baseentity.cpp
+            if (angle > 86 && angle < 170)
+                damage = damage * (1.00 + PChar->PMeritPoints->GetMeritValue(MERIT_INNIN_EFFECT, PChar)/100.0);
```

Comment:
Sorry for this. But 1.00 and 100.0 are doubles. Use 1.f and 100.f for consistency and no need for 64bit floating point.

Reply:
I promise that’s the last thing.

Reply:
I just looked at the new build failure. You need to explicitly cast the result to int32.

Reply:
If I build via site instructions and instead of CMake, would I see these errors pop up? Or is this just an appveyor thing?

Reply:
travis tests the ubuntu build and appveryor tests the windows build. basically your code doesn't work in visual studio's compiler.

Reply:
Seems like cmake don't use `-Werror`/"Warnings as Errors" flags when generating the solution where-as the provided ones do, hence the difference.

Reply:
Terrible, but understood.

Reply:
Hmmm, wait @zynjec , are you sure? The only warnings are I see when compiling are generated by code that is not mine.

Reply:
```
C:\projects\darkstar\src\map\utils\battleutils.cpp(3105): warning C4244: '=': conversion from 'double' to 'int32', possible loss of data [C:\projects\darkstar\src\map\DSGame-server.vcxproj]
```

It's the line Autkast started the comment on.

Reply:
Yah, fixed and pushed.